File: libs/shared-auth/src/main/java/br/com/deltaglobalbank/sharedauth/AuthenticatedPrincipal.java
package br.com.deltaglobalbank.sharedauth;

import java.util.List;
import java.util.UUID;

public record AuthenticatedPrincipal(
    UUID subject,
    UUID tenantId,
    String principalType,
    List<String> roles,
    List<String> modules,
    boolean mustChangePassword,
    UUID jti
) {
    public boolean isUser() {
        return "user".equals(principalType);
    }

    public boolean isApiClient() {
        return "api_client".equals(principalType);
    }
}

File: libs/shared-auth/src/main/kotlin/br/com/deltaglobalbank/sharedauth/AuthenticatedPrincipal.kt
package br.com.deltaglobalbank.sharedauth

import java.util.UUID

data class AuthenticatedPrincipal(
    val subject: UUID,
    val tenantId: UUID,
    val principalType: String,
    val roles: List<String>,
    val modules: List<String>,
    val mustChangePassword: Boolean,
    val jti: UUID
) {
    val isUser: Boolean get() = principalType == "user"
    val isApiClient: Boolean get() = principalType == "api_client"
}
